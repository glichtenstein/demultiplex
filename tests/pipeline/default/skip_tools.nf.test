nextflow_pipeline {

    name "Test skips"
    script "main.nf"

    test("Should skip trimming") {

        when {
            params {
                input = "$baseDir/assets/inputs/flowcell_input.csv"
                demultiplexer = 'bclconvert'
                trim_fastq  = false
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 7

            assert snapshot(
                path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz"),
                path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz.md5"),
                path("$outputDir/220422_M11111_0222_000000000-K9H97/Undetermined_S0_L001_R1_001.fastq.gz"),
                path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
            ).match()
        }

    }

}
