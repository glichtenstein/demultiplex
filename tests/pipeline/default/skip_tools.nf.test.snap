{
    "Should skip trimming": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0a0341e2990b4fa1d9ad4b4c603144c1",
            "Sample1_S1_L001_R1_001.fastq.gz.md5:md5,c547a3c03e7c60c871d78018ec5d933f",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,febef808ae5397ea4ee7ee40e5fd39e0",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,ee5db2e12754e069998b0a96e535238c",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
            ]
        ],
        "timestamp": "2023-01-18T22:31:23+0000"
    }
}